File: frontend/src/app/classes/RendezVous.ts
import { Time } from "@angular/common"
import { StatusRDV } from "./enums/StatusRDV"
import { patient } from "./patient"

export class RendezVous{
    dateRendezVous!:Date
    heureDebut!:Time
    heureFin!:Time
    status!:StatusRDV
    patient!:patient
}

File: gestionCabinet/src/main/java/com/example/demo/entities/RendezVous.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

import com.example.demo.enums.StatusRDV;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "rendezvous")
public class RendezVous {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Temporal(TemporalType.DATE)
    private Date dateRendezVous;
    private String heureDebut;
    @Enumerated(EnumType.STRING)
    private StatusRDV status;
    @ManyToOne
    private Patient patient;

}
